R/convGDX2MIF.R

```diff
@@ -1,128 +1,172 @@
+#'     scenario for all time steps before `cm_startyear`.
+#' @param file Name of the .mif file which will be written.  If no name is
+#'     provided a magpie object containing all the reporting information is
+#'     returned.
+#' @param scenario Scenario name that is used in the *.mif reporting.  Defaults
+#'     to `'default'`.
+#' @param t Temporal resolution of the reporting, defaults to
+#'     `t = c(seq(2005, 2060, 5), seq(2070, 2110, 10), 2130, 2150)`.
+#' @param gdx_refpolicycost Reference gdx file path for policy costs.
 #'
-#' @param gdx a GDX as created by readGDX, or the file name of a gdx
-#' @param gdx_ref reference-gdx for < cm_startyear, used for fixing the prices to this scenario
-#' @param file name of the mif file which will be written, if no name is
-#' provided a magpie object containing all the reporting information is
-#' returned
-#' @param scenario scenario name that is used in the *.mif reporting
-#' @param t temporal resolution of the reporting, default:
-#' t=c(seq(2005,2060,5),seq(2070,2110,10),2130,2150)
-#' @param gdx_refpolicycost reference-gdx for policy costs, a GDX as created by readGDX, or the file name of a gdx
 #' @author Lavinia Baumstark
-#' @examples
 #'
-#' \dontrun{convGDX2MIF(gdx,gdx_refpolicycost,file="REMIND_generic_default.csv",scenario="default")}
+#' @examples
+#' \dontrun{
+#'     convGDX2MIF(gdx, gdx_refpolicycost, file = "REMIND_generic_default.csv",
+#'                 scenario = "default")
+#' }
 #'
-#' @export
 #' @importFrom gdx readGDX
 #' @importFrom magclass mbind write.report
+#' @importFrom methods formalArgs getFunction
 
+#' @export
 convGDX2MIF <- function(gdx, gdx_ref = NULL, file = NULL, scenario = "default",
-                        t = c(seq(2005, 2060, 5), seq(2070, 2110, 10), 2130, 2150),
+                        t = c(seq(2005, 2060, 5), seq(2070, 2110, 10), 2130,
+                              2150),
                         gdx_refpolicycost = gdx_ref) {
 
   # Define region subsets
   regionSubsetList <- toolRegionSubsets(gdx)
+
   # ADD EU-27 region aggregation if possible
-  if("EUR" %in% names(regionSubsetList)){
-    regionSubsetList <- c(regionSubsetList,list(
-      "EU27"=c("ENC","EWN","ECS","ESC","ECE","FRA","DEU","ESW")
-    ))
+  if ("EUR" %in% names(regionSubsetList)) {
+    regionSubsetList <- c(regionSubsetList,
+                          list("EU27" = c("ENC", "EWN", "ECS", "ESC", "ECE",
+                                          "FRA", "DEU", "ESW")))
   }
 
-  # make the reporting
-  output <- NULL
-  message("running reportMacroEconomy...")
-  output <- mbind(output,reportMacroEconomy(gdx,regionSubsetList,t)[,t,])
-  message("running reportTrade...")
-  output <- mbind(output,reportTrade(gdx,regionSubsetList,t)[,t,])
-  message("running reportPE...")
-  output <- mbind(output,reportPE(gdx,regionSubsetList,t)[,t,])
-  message("running reportSE...")
-  output <- mbind(output,reportSE(gdx,regionSubsetList,t)[,t,])
-  message("running reportFE...")
```

Comment:
I think these messages are helpful to see the progress (at least when running reporting locally) and should not be removed.

Reply:
> see the progress

As in "is this running or did this crash"?
I was [under the impression](https://github.com/pik-piam/remind2/commit/218738e369fe73123939731ce1a65fc04e9c53d8) this was added to figure out where it crashed if/when it crashed, which is now handled explicitly.
Anyhow, added messages (that can be turned off).